mlir/include/mlir/Dialect/GPU/IR/GPUOps.td

```diff
@@ -3255,4 +3255,37 @@ def GPU_SubgroupBroadcastOp : GPU_Op<"subgroup_broadcast",
   let hasVerifier = 1;
 }
 
+def GPU_SubgroupUniformOp : GPU_Op<"subgroup_uniform",
+    [Pure, AllTypesMatch<["result", "src"]>,
```

Comment:
```suggestion
    [NoMemoryEffect, AllTypesMatch<["result", "src"]>,
```

It's not clear to me that this is all speculatable with respect to the property it provides?